core/services/ocr2/plugins/ocr2keeper/evm/abi.go

```diff
@@ -9,13 +9,19 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/smartcontractkit/ocr2keepers/pkg/types"
+
+	"github.com/smartcontractkit/ocr2keepers/pkg/chain"
 )
 
-type evmRegistryPackerV2_0 struct {
+type EvmRegistryPackerV2_0 struct {
```

Comment:
had to make this visible outside this package. Not sure if ideal, we need to think more about structuring our packages anyway

Reply:
The only reason to make it visible is because you used a reference directly instead of an interface. This breaks the abstraction of different packers per chain.

In `log_provider.go` create the following interface:
```
type TransmitUnpacker interface {
    UnpackTransmitTxInput([]byte) ([]types.UpkeepResult, error)
}
```

And then set `LogProvider.packer` to the type of `TransmitUnpacker`. Then you can pass the result of `NewEvmRegistryPackerV2_0` to `NewLogProvider` as a dependency without exposing the underlying registry packer struct.

From there, the dependency injection happens in the `ocr2keeper` package, which allows for a future selection of evm vs anything else.